### PR TITLE
ExplicitNonNullaryApply also fix ApplyType

### PR DESCRIPTION
**Pull request by ohze**
Given:
```
object Props {
  def apply[A]() = ???
}
```
Then `Props[Int]` is rewrite to `Props[Int]()`

### CODE REVIEW

File: rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala [hunk 1]
```diff
@@ -45,6 +45,13 @@ final class ExplicitNonNullaryApply(global: LazyValue[ScalafixGlobal])
         if !power.isJavaDefined(name) // !info.isJava
         if cond(info.signature) {
           case MethodSignature(_, List(Nil, _*), _) => true
+          case ClassSignature(_, _, _, decls) if tree.isInstanceOf[Term.ApplyType] =>
+            decls.exists { decl =>
+              decl.displayName == "apply" &&
```
Comment (dwijnand) [inline rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala:50]: I think we can drop this one, as it wouldn't compile other, no?

Comment (ohze) [reply rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala:50]: No. Given
```scala
object Props {
    def apply[A]() = ??? // (1)
    def foo() = ???          // (2)
}
```
We want fix `Props[Int]` to `Props[Int]()` but only if (1) is non-nullary.
+ `cond(decl.signature) { ... ` condition to check `decl` is a non-nullary method
+ `decl.displayName == "apply"` condition to check `decl` is (1), not another method like (2)

Comment (ohze) [reply rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala:50]: That means:
+ Without `decl.displayName == "apply"` then we will incorrectly add parens to `Props[Int]` for
```scala
object Props {
    def apply[A] = ??? // nullary!
    def foo() = ??? 
}
``` 
+ Without `cond(decl.signature) { ...` then we will incorrectly add parens to `Props[Int]` for
```scala
object Props {
    def apply[A] = ???
}
```

**Comment by ohze** [comment]
I have just update the test code to cover cases describe in https://github.com/scala/scala-rewrites/pull/25#discussion_r414036841 (the comment above)

File: rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala (hunk 1 above)
Comment (dwijnand) [reply rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala:50]: Ah, I see!

**Comment by dwijnand** [comment]
> I have just update the test code to cover cases describe in [#25 (comment)](https://github.com/scala/scala-rewrites/pull/25#discussion_r414036841) (the comment above)

Wonderful.  Thank you.

**Pull request merged**